packages/engine/src/datastore/table/state/mod.rs

```diff
@@ -264,39 +208,48 @@ impl StateMut {
 
     // TODO: enable writing into the message batch too
     pub fn set_pending_column(&mut self, column: StateColumn) -> Result<()> {
-        self.inner.agent_pool.set_pending_column(column)
+        self.agent_pool_mut().set_pending_column(column)
     }
 
     pub fn flush_pending_columns(&mut self) -> Result<()> {
-        self.inner.agent_pool.flush_pending_columns()
+        self.agent_pool_mut().flush_pending_columns()
     }
-}
 
-impl ReadState for StateMut {
-    fn inner(&self) -> &Inner {
-        &self.inner
+    pub fn sim_config(&self) -> &Arc<SimRunConfig> {
+        &self.sim_config
     }
 
-    fn sim_config(&self) -> &Arc<SimRunConfig> {
-        &self.global_meta
+    pub fn message_pool(&self) -> &MessagePool {
+        &self.message_pool
     }
-}
 
-impl WriteState for StateMut {
```

Comment:
I guess the intention with `WriteState` was that it could be applied to more than just a `State` object? So that some `PartialState` or something could also implement `WriteState`.

Not a bad design if that was the case, but it doesn't seem like we actually need it anymore so I agree with removing it unless we think that keeping it and using it as intended could improve the current situation.

Reply:
`ReadState` and `WriteState` was only used for a unified API for `State` and `StateMut`. Actually, `WriteState` was only implemented for `StateMut`.

Reply:
I guess I could imagine the Rust runner using it, but I'm not sure it will and it's not essential, so we can remove it.